src/cljs/org/broadinstitute/firecloud_ui/utils.cljs

```diff
@@ -94,34 +99,34 @@
-                                                 user-info (keywordize-keys parsed-us-response)]
-                                                (cond
-                                                  (and (= us-status 200) (not (:ldap (:enabled (:userInfo user-info)))))
-                                                  (on-done {:xhr us-xhr
-                                                            :status-code 403
-                                                            :success? false
-                                                            :status-text "Access Disabled"
-                                                            :get-parsed-response parsed-us-response})
-                                                  ;TODO: Fix this with a real log-out once the login bug is fixed and logout is implemented.
-                                                  (= us-status 401) (set! (-> js/window .-location) "/")
-                                                  :else (on-done {:xhr us-xhr
-                                                                  :status-code us-status
-                                                                  :success? false
-                                                                  :status-text (.-statusText us-xhr)
-                                                                  :get-parsed-response parsed-us-response})))))
-                                     (.open us-xhr "GET" "/service/register")
-                                     (.setRequestHeader us-xhr "Authorization" (str "Bearer " @access-token))
-                                     (.setRequestHeader us-xhr "Content-Type" "application/json" )
-                                     (.setRequestHeader us-xhr "Accept" "application/json" )
-                                     (.send us-xhr))
-                              (on-done {:xhr xhr
+                           (if (= status-code 401)
+                             (let [us-xhr (js/XMLHttpRequest.)]
+                               (set! (.-withCredentials us-xhr) true)
+                               (.addEventListener us-xhr "loadend"
+                                 (fn []
+                                   (let [us-status (.-status us-xhr)
+                                         parsed-us-response (parse-json-string (.-responseText us-xhr))
+                                         user-info (keywordize-keys parsed-us-response)]
+                                     (cond
+                                       (and (= us-status 200) (not (:ldap (:enabled (:userInfo user-info)))))
+                                       (on-done {:xhr (adjust-xhr xhr)
+                                                 :status-code 403
+                                                 :success? false
+                                                 :status-text "Access Disabled"
+                                                 :get-parsed-response get-parsed-response})
```

Comment:
I'm not following this. In this case, `get-parsed-response` is always the generic 401 Unauthorized message. I think it makes more sense to return the response from the secondary user-status request because that has real actionable information. The 401 response can sometimes come from the proxy and is pretty much useless.